language/src/bin.rs

```diff
@@ -49,12 +49,73 @@ fn main() {
 
     let environment: HashMap<String, String> = env::vars().collect();
 
+    let file_paths = match args.iter().position(|a| a == "-o") {
+        Some(j) => {
+            match args.get(j + 1).cloned() {
+                Some (file) => {
+                    let file_parent = std::path::Path::new(&file).parent().and_then(|x| std::ffi::OsStr::to_str(x.as_os_str())).unwrap();
+                    let file_stem = std::path::Path::new(&file).file_stem().and_then(std::ffi::OsStr::to_str).unwrap();
+                    let file_extension = std::path::Path::new(&file).extension().and_then(std::ffi::OsStr::to_str).unwrap();
+                    
+                    let template_path = file_parent.to_string() + "/" + file_stem + "_template." + file_extension;
```

Comment:
Please use `path` here as well so you don't have to hard code the separator `/`.